docs-ui/components/code.tsx

```diff
@@ -0,0 +1,142 @@
+import {withTheme} from '@emotion/react';
+import styled from '@emotion/styled';
+import copy from 'copy-text-to-clipboard';
+import Prism from 'prismjs';
+import 'prismjs/themes/prism.css';
+import {Component, createRef, RefObject, useEffect} from 'react';
+
+import {IconCode} from 'app/icons';
+import space from 'app/styles/space';
+import {Theme} from 'app/utils/theme';
+
+type Props = {
+  theme: Theme;
+  /**
+   * Main code content gets passed as the children prop
+   */
+  children: string;
```

Comment:
Should this be `React.ReactNode` so it could be a bit more flexible?

Reply:
Good point. Just tried inserting some non-text elements as children, and Prism converted all of them into plain text (<button>Hello</button> becomes 'Hello'), so I don't expect people to do this. Still, we should probably use `React.ReactNode` to account for when they do.

Reply:
It might make some other parts of typing harder. So I would be OK requiring it to be a string

Reply:
Now that I see what this is doing, I think it's OK